superset-frontend/plugins/plugin-chart-echarts/src/utils/buildCustomFormatters.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  Currency,
+  CurrencyFormatter,
+  ensureIsArray,
+  getNumberFormatter,
+  isDefined,
+  isSavedMetric,
+  NumberFormatter,
+  QueryFormMetric,
+} from '@superset-ui/core';
+
+export const buildCustomFormatters = (
+  metrics: QueryFormMetric | QueryFormMetric[] | undefined,
+  currencyFormats: Record<string, Currency>,
+  columnFormats: Record<string, string>,
+  d3Format: string | undefined,
+  labelMap: Record<string, string[]> = {},
+  seriesNames: string[] = [],
+) => {
```

Comment:
It seems the return type is `{ [key: string]: CurrencyFormatter | NumberFormatter }`, could we add that explicitly here to make it easier to read?

Reply:
Good point